src/classes/User.js

```diff
@@ -0,0 +1,43 @@
+class User {
+  constructor(name, id) {
+    this.name = name;
+    this.id = id;
+    this.favoriteRecipes = [];
+    this.recipesToCook = [];
+  }
+
+  addToFavorites(recipe) {
+    if (!this.favoriteRecipes.includes(recipe)) {
+      this.favoriteRecipes.push(recipe);
+    }
+  }
+
+  removeFromFavorites(recipe) {
+    this.favoriteRecipes = this.favoriteRecipes.filter(favRecipe => {
+      if (favRecipe.id !== recipe.id) {
+        return favRecipe;
+      }
+    });
+  }
+
+  addToRecipesToCook(recipe) {
+    if (!this.recipesToCook.includes(recipe)) {
+      this.recipesToCook.push(recipe);
+    }
+  }
+
+  filterByTag(tag) {
```

Comment:
hmm would we be able to use the RecipeCollection method filterByTag here or no? would probably make it way messier

Reply:
In a future iteration we could extend a class and reuse your method from RecipeCollection!